lib/difi_sink_cpp_impl.cc

```diff
@@ -92,14 +90,24 @@ namespace gr {
       pack_u32(&d_context_raw[0], tmp_header_context);
       pack_u32(&d_context_raw[4], d_stream_number);
       idx = 0;
-      pack_u64(&d_context_raw[difi::CONTEXT_PACKET_OFFSETS[idx++]], class_id);
+      pack_u64(&d_context_raw[difi::CONTEXT_PACKET_OFFSETS[idx++]], d_context_class_id);
       // this is static since only 8 or 16 bit signed complex cartesian is supported for now and item packing is always link efficient
       // see 2.2.2 Standard Flow Signal Context Packet in the DIFI spec for complete information
       u_int64_t data_payload_format = bit_depth == 8 ? difi::EIGHT_BIT_SIGNED_CART_LINK_EFF : difi::SIXTEEN_BIT_SIGNED_CART_LINK_EFF;
 
       u_int32_t state_and_event_id =difi::DEFAULT_STATE_AND_EVENTS; // default no events or state values. See 7.1.5.17 The State and Event Indicator Field of the VITA spec
       u_int64_t to_vita_bw = u_int64_t(samp_rate * .8) << 20; // no fractional bw or samp rate supported in gnuradio, see 2.2.2 Standard Flow Signal Context Packet for bandwidth information
+      u_int64_t to_vita_if_band_offset= int64_t(samp_rate * -0.1) << 20; //generic IF band offset (1Hz resolution)
       u_int64_t to_vita_samp_rate = samp_rate << 20;
+      u_int64_t to_vita_if_ref_freq = 100000000UL << 20; //generic 100MHz IF reference frequency
```

Comment:
is the idea for all these fake values to eventually add block params to fill them out?  also it makes me wonder if it's better to provide a fake value vs just 0 to indicate it wasn't filled out (works for some params but not all)

Reply:
For certification, all values are fake but nonzero values catch more bugs.

Eventually, yes, adding more block parameters is the way.